.github/workflows/build-and-upload-archives-upon-creating-tags.yml

```diff
@@ -24,11 +24,152 @@ on:
       - '*'
 
 jobs:
-  build:
+  code-analysis:
     # Name the Job
-    name: Build tagged commit and upload an archive
+    name: Code analysis
     # Set the type of machine to run on
     runs-on: ubuntu-latest
+    steps:
+      # Checks out a copy of your repository on the ubuntu-latest machine
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          # bring in all history because the gradle versions plugin needs to "walk back" to the closest ancestor tag
+          fetch-depth: 0
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: '11'
+      - name: Run code analysis tasks
+        run: ./gradlew rat checkstyleMain checkstyleTest spotbugsMain spotbugsTest --no-daemon -PxmlSpotBugsReport=true -PtestLoggingEvents=started,passed,skipped,failed
+
+  unit-test:
```

Comment:
Can you please check if we can leverage reusable workflows to avoid code duplication?
https://docs.github.com/en/actions/using-workflows/reusing-workflows

If so, this workflow could call the unit-test and int-test workflows. Thank you.